src/components/UpdatePassword/UpdatePasswordButton.tsx

```diff
@@ -0,0 +1,136 @@
+import { useContext, useRef, useState } from "react";
+import { updatepasswordService } from "../../services/auth/updatepassword.service";
+import { FieldValues, useForm } from "react-hook-form";
+import { AuthContext } from "../../context/AuthContext";
+import toast from "react-hot-toast";
+
+export function UpdatePassword() {
+  const { session } = useContext(AuthContext);
+  const [showModal, setShowModal] = useState(false);
+  const formRef = useRef<HTMLFormElement>(null);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm();
+
+  interface UpdatepasswordRequest {
+    oldPassword: string;
+    newPassword: string;
+    token: string;
+  }
+
+  const onSubmit = async (formData: FieldValues, token: string) => {
+    const req: UpdatepasswordRequest = {
+      oldPassword: formData.password,
+      newPassword: formData.newPassword,
+      token: token
+    };
+    if (!errors.password && !errors.newPassword) {
+      try {
+        const response = await updatepasswordService(req);
+        if (response.success === true) {
+          toast.success("Password updated successfully");
+          setShowModal(false);
+        } else {
+          toast.error("Failed to update password");
+        }
+      } catch (error) {
+        toast.error("Failed to update password");
+      }
+      formRef.current?.reset();
+    }
+  };
+
+  return (
+    <>
+      <button
+        className="w-full rounded-lg border border-blue-600 bg-white p-2 text-center text-lg font-bold text-blue-600"
+        onClick={() => setShowModal(true)}
+      >
+        Update Password
+      </button>
+      {showModal ? (
+        <div className="fixed inset-0 z-10 flex items-center justify-center backdrop-blur-sm">
```

Comment:
You copied the `Dialog` component styles, use the `Dialog` component itself then.

Reply:
the title is in white color but I assume it was changed in later PRs
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/e1ac2ce8-9beb-4af7-9521-b31e1377afd0)

Reply:
@DanielEJaimes no, it wasn't. It is because of the `text-white` in the `Nav.tsx` file. Consider adding the `text-black` or `text-blue-600` `className` to the title in the `Dialog.tsx` file.

Reply:
Made it blue-600